src/tools/id-mapping/components/IDMappingForm.tsx

```diff
@@ -0,0 +1,465 @@
+      for (const [key, field] of defaultValuesEntries) {
+        formValues[key] = Object.freeze({
+          ...field,
+          selected:
+            parametersFromHistoryState[
+              field.fieldName as keyof FormParameters
+            ] || field.selected,
+        }) as Readonly<IDMappingFormValue>;
+      }
+      return Object.freeze(formValues) as Readonly<IDMappingFormValues>;
+    }
+    // otherwise, pass the default values
+    return defaultFormValues;
+  }, [history]);
+
+  // actual form fields
+  const initialIDs = initialFormValues[IDMappingFields.ids]
+    .selected as string[];
+  // Text of IDs from textarea
+  const [textIDs, setTextIDs] = useState<string>(joinIDs(initialIDs));
+  // Parsed (by whitespace) IDs
+  const [ids, setIDs] = useState<string[]>(initialIDs);
+  const [fromDb, setFromDb] = useState<IDMappingFormValue>(
+    initialFormValues[IDMappingFields.fromDb]
+  );
+  const [toDb, setToDb] = useState<IDMappingFormValue>(
+    initialFormValues[IDMappingFields.toDb]
+  );
+  const [taxID, setTaxID] = useState<IDMappingFormValue>(
+    initialFormValues[IDMappingFields.taxons]
+  );
+
+  // extra job-related fields
+  const [jobName, setJobName] = useState(
+    initialFormValues[IDMappingFields.name]
+  );
+  // flag to see if the user manually changed the title
+  const [jobNameEdited, setJobNameEdited] = useState(false);
+
+  const [submitDisabled, setSubmitDisabled] = useState(false);
+  const [sending, setSending] = useState(false);
+
+  const { loading, data, error } = useDataApi<IDMappingFormConfig>(
+    apiUrls.idMappingFields
+  );
+
+  const [dbNameToDbInfo, ruleIdToRuleInfo]: [
+    DbNameToDbInfo | undefined | null,
+    RuleIdToRuleInfo | undefined | null
+  ] = useMemo(() => {
+    if (!data) {
+      return [null, null];
+    }
+    let dbNameToDbInfo;
+    let ruleIdToRuleInfo;
+    if (data) {
+      dbNameToDbInfo = Object.fromEntries(
+        data.groups.flatMap(({ items }) =>
+          items.map((item) => [item.name, item])
+        )
+      );
+      ruleIdToRuleInfo = Object.fromEntries(
+        data.rules.map((rule) => [rule.ruleId, rule])
+      );
+    }
+    return [dbNameToDbInfo, ruleIdToRuleInfo];
+  }, [data]);
+
+  const submitIDMappingJob = useCallback(
+    (event: FormEvent | MouseEvent) => {
+      event.preventDefault();
+
+      if (!ids.length || !dbNameToDbInfo || !ruleIdToRuleInfo) {
+        return;
+      }
+
+      setSubmitDisabled(true);
+      setSending(true);
+
+      const fromDbInfo = dbNameToDbInfo[fromDb.selected as string];
+      const ruleInfo =
+        fromDbInfo?.ruleId && ruleIdToRuleInfo[fromDbInfo.ruleId];
+
+      // here we should just transform input values into FormParameters,
+      // transformation of FormParameters into ServerParameters happens in the
+      // tools middleware
+      const parameters: FormParameters = {
+        from: fromDb.selected as string,
+        to: toDb.selected as string,
+        ids,
```

Comment:
OK, so have a think if you want to pass here the cleaned input, or the raw input. If you pass the raw input, the cleaning can be done again inside the `formParametersToServerParameters` function.

Reply:
If you do use the raw, it would mean `FormParameters['ids']` needs to be defined as `string` instead of `string[]`

Reply:
But as I said, it's not a requirement for me, it's up to you